### ISSUE TITLE
Cannot use Selendroid's "Testing the Mobile App" feature through Appium

### ISSUE THREAD

**Issue by vbanthia** [appium/appium#5411]
Hello guys,

Selendroid has a feature of testing mobile app using a hybrid app. In this selendroid installs "io.selendroid.androiddriver" apk and let user to test their mobile web app using this hybrid app.
http://selendroid.io/mobileWeb.html

I was trying following capabilities to get this run using Appium

Trial 1

``` ruby
require 'selenium-webdriver'

driver_caps = {
  'platformName' => 'Android',
  'automationName' => 'Selendroid', 
  'browserName' => 'Browser'
  'deviceName' => '',
}
driver = Selenium::WebDriver.for(:remote, desired_capabilities: driver_caps, url: url)
```

But this was raising below error

```
UiAutomator quit before it successfully launched
```

I debugged and found that if browserName is set in capabilities then it will create chrome_driver instance instead of selendroid
Source: https://github.com/appium/appium/blob/master/lib/appium.js#L175-L185

So, in my next try I removed browserName and used only automationName

Trial 2

``` ruby
require 'selenium-webdriver'

driver_caps = {
  'platformName' => 'Android',
  'automationName' => 'Selendroid', 
  'deviceName' => '',
}
driver = Selenium::WebDriver.for(:remote, desired_capabilities: driver_caps, url: url)
```

But this was raising no app set error.

```
No app set; either start appium with --app or pass in an 'app' value in desired capabilities, or set androidPackage to launch pre-existing app on device
```

Below is the related code

``` node
androidCommon.configure = function (args, caps, cb) {
  this._deviceConfigure(args, caps);
  this.setAndroidArgs();
  if (!this.args.androidActivity) {
    logger.debug("No appActivity desired capability or server param. " +
                "Parsing from apk.");
  }
  if (!this.args.androidPackage) {
    logger.debug("No appPackage desired capability or server param. " +
                "Parsing from apk.");
  }

  if (this.args.app) {
    this.configureApp(cb);
  } else if (this.args.androidPackage) {
    this.args.app = null;
    logger.debug("Didn't get app but did get Android package, will attempt " +
                "to launch it on the device");
    cb(null);
  } else {
    var msg = "No app set; either start appium with --app or pass in an " +
              "'app' value in desired capabilities, or set androidPackage " +
              "to launch pre-existing app on device";
    logger.error(msg);
    cb(new Error(msg));
  }
};
```

As can be seen in the code if (app | androidPackage) is not set then this will raise error. But for selendroid mobile-web app case both app and androidPackage are not required.

**Comment by imurchie** [comment]
I'm not sure what the use case of this is. Why do you not want to run your web application in a browser?

**Comment by vbanthia** [comment]
@imurchie 
The use case is for android 2.3.x. This is the only way to test mobile web app for 2.3.x. Appium supports web browser for only 4.2+.

Note:
I came through this problem while researching about Appium + Selendroid. I do not have any practical use case. So, it is not a blocker for me. I reported this because it was not mentioned in Appium documents and there were other people who were also facing this problem like [this guy](https://discuss.appium.io/t/testing-mobile-web-app-in-appium-with-selendroid/1127).
